src/shims/env.rs

```diff
@@ -23,8 +24,12 @@ pub struct EnvVars<'tcx> {
 impl<'tcx> EnvVars<'tcx> {
     pub(crate) fn init<'mir>(
         ecx: &mut InterpCx<'mir, 'tcx, Evaluator<'tcx>>,
-        excluded_env_vars: Vec<String>,
+        mut excluded_env_vars: Vec<String>,
     ) -> InterpResult<'tcx> {
+        if ecx.tcx.sess.target.target.target_os.to_lowercase() == "windows" {
```

Comment:
Do we use `to_lowercase` for this anywhere in the compiler?
I would've thought we didn't

Reply:
> 
> 
> Do we use `to_lowercase` for this anywhere in the compiler?
> I would've thought we didn't

I also never saw using `to_lowercase` for this elsewhere in the compiler.
I naively copied that line from [a previous commit](https://github.com/rust-lang/miri/commit/39146c44bde0950a782446c9e4d95f8071acef54#diff-6d61240795974400d2a4ee29e5c02d39L24). I can remove the `to_lowercase()` invocation.

Reply:
We used to do `to_lowercase` here in Miri but stopped doing so a while ago. The patch @JOE1994 copied this from is from before that.

But yeah, this should be just `to_str` or so now, like elsewhere.

Reply:
Elsewhere doesn't use `.as_str()` at all, you can just compare it to a string literal.

Reply:
Ah when that works, even better.

Reply:
It's not entirely true though -- when `match` on the result, as need `as_str()` or `&<expr>[..]`. A plain `&` does not suffice.

Reply:
Yes, `==` is what I mean by "compare", pattern-matching is one of the few cases in which coercions don't trigger (the other one is calls to generic functions where there's nothing hinting that a `&str` is expected).